pkg/fqdn/proxy/proxy.go

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2016-2020 Authors of Cilium
+
+package proxy
+
+import (
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/fqdn/restore"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+type FQDNProxy interface {
```

Comment:
Nit, I suspect you're trying to avoid naming this `DNSProxy` since that's the concrete implementation, but the function of the implementer here is to proxy DNS packets. Fully Qualified Domain Name Proxy doesn't really make sense, the proxy proxies requests/responses, not names. I wonder if we could just take the verb form similar to some of the base golang interfaces?

```suggestion
type DNSProxier interface {
```

Reply:
+1